frontend/src/static/js/main.js

```diff
@@ -155,6 +156,37 @@ window.app = new window.Vue({
       }
     },
 
+    renderAuthorShipTab() {
+      const encodedInfo = window.location.hash.slice(1).split('&');
+      this.deserialize(encodedInfo);
+      const hash = window.hashParams;
+      if (hash.info) {
+        this.updateTabAuthorship(hash.info);
+      } else if (hash.isTabOpen === false) {
+        this.isTabActive = false;
+      }
+    },
+
+    deserialize(obj) {
+      const info = {};
+      obj.forEach((param) => {
+        const [key, val] = param.split('=');
+        if (key === 'info') {
+          const decodedKey = decodeURIComponent(val);
+          const properties = decodedKey.split('&');
+          properties.forEach((prop) => {
+            const [innerKey, innerVal] = prop.split('=');
+            info[innerKey] = decodeURIComponent(innerVal);
+          });
+        } else if (key === 'isTabOpen') {
+          window.hashParams[key] = (val === 'true');
+        }
+      });
+      if (Object.keys(info).length > 0) {
+        window.hashParams.info = info;
```

Comment:
updates on `window.hashParams` should be on its own. look at lin 182 from `v_summary`, shift that function out as well.

make a vanilla function in `main.js` called `updateHash()` then your `v_summary.renderFilterHash` and your `renderAuthorship` can just start with `const hash = window.hashParams`